src/plugins/chart_expressions/expression_heatmap/public/components/heatmap_component.tsx

```diff
@@ -286,26 +286,29 @@ export const HeatmapComponent: FC<HeatmapRenderProps> = memo(
 
     const bands = ranges.map((start, index, array) => {
       // by default the last range is right-open
-      let end = index === array.length - 1 ? Number.POSITIVE_INFINITY : array[index + 1];
+      let endValue = index === array.length - 1 ? Number.POSITIVE_INFINITY : array[index + 1];
```

Comment:
nit: I am wondering if readability can be improved by some minor refactoring:

```ts
const hasSingleValue = max === min;
const bands = ranges.map((start, index, array) => {
      const isPenultimate = index === array.length - 1;
      const nextValue = array[index + 1];
      // by default the last range is right-open
      let endValue = isPenultimate? Number.POSITIVE_INFINITY : nextValue;
      const startValue = isPenultimate && hasSingleValue ? min : start;
      // if the lastRangeIsRightOpen is set to false, we need to set the last range to the max value
      if (args.lastRangeIsRightOpen === false) {
        const lastBand = hasSingleValue ? Number.POSITIVE_INFINITY : endValueDistinctBounds;
        endValue = isPenultimate ? lastBand : nextValue;
      }

      let overwriteArrayIdx;

      if (endValue === Number.POSITIVE_INFINITY) {
        overwriteArrayIdx = `≥ ${metricFormatter.convert(startValue)}`;
      } else {
        overwriteArrayIdx = `${metricFormatter.convert(start)} - ${metricFormatter.convert(
          endValue
        )}`;
      }

      const overwriteColor = overwriteColors?.[overwriteArrayIdx];
      return {
        // with the default continuity:above the every range is left-closed
        start: startValue,
        end: endValue,
        // the current colors array contains a duplicated color at the beginning that we need to skip
        color: overwriteColor ?? colors[index + 1],
      };
```

what do you think?

Reply:
I love it Marco :)